src/runtime/refiner.ts

```diff
@@ -53,13 +53,23 @@ export class Refinement {
   constructor(expr: RefinementExpression) {
     // TODO(ragdev): Should be a copy?
     // TODO(ragdev): ensure that the refinement contains at least 1 fieldName
+    if (!expr) {
+      return null;
+    }
     this.expression = expr;
   }
 
   static fromAst(ref: RefinementNode, typeData: Dictionary<ExpressionPrimitives>): Refinement {
     return new Refinement(RefinementExpression.fromAst(ref.expression, typeData));
   }
 
+  static fromLiteral(ref): Refinement {
+    if (!ref) {
```

Comment:
I think this is a call site concern, I'd suggest we move the checks and returns out of these three functions.